backend/internal/auth/firebase.go

```diff
@@ -0,0 +1,18 @@
+package auth
+
+import (
+	"context"
+	"firebase.google.com/go/v4/auth"
+)
+
+type FirebaseAuthenticator struct {
+	client auth.Client
+}
+
+func (a *FirebaseAuthenticator) VerifyToken(ctx context.Context, authHeader string) (string, bool) {
```

Comment:
Documentation on this implementation would be helpful (mention that it verifies the JWT is signed and is valid for the Firebase project)

Reply:
Also, this function does *not* take the entire Authorization header, it just takes the token. Let's rename `authHeader` to `token`